cmd/meshregistrator/pods.go

```diff
@@ -0,0 +1,125 @@
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"io/ioutil"
+	"net/http"
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+	klog "k8s.io/klog/v2"
+)
+
+const HacheckPodName = "hacheck"
+
+// fetch running pods from kubelet and update pods registration list
+func fetchPods(ctx context.Context, out chan []ServiceRegistration) {
+	var oldRegistrations, newRegistrations []ServiceRegistration
+	ticker := time.NewTicker(5 * time.Second)
+	defer ticker.Stop()
+
+	for {
+		select {
+		case <-ctx.Done():
+			klog.Info("fetchPods stopped")
+			return
+		case <-ticker.C:
+			startTime := time.Now().UnixNano()
+			resp, err := http.Get("http://127.0.0.1:10255/pods")
```

Comment:
I'm surprised you don't need creds to do this?

Reply:
not for local kubelet api apparently, same code in paasta-tools